registration-processor/core-processor/registration-processor-uin-generator-stage/src/test/java/io/mosip/registration/processor/stages/uigenerator/UinGeneratorStageTest.java

```diff
@@ -1,2164 +1,2473 @@
+		idResponseDTO.setResponse(null);
+		idResponseDTO.setResponsetime("2019-03-12T06:49:30.779Z");
+		idResponseDTO.setVersion("1.0");
+
+		// Mockito.when(packetReaderService.getFile("10031100110005020190313110030",
+		//// PacketFiles.ID.name(), "id"))//.thenReturn(idJsonStream1);
+		ApisResourceAccessException ex=new ApisResourceAccessException("", new HttpServerErrorException(HttpStatus.INTERNAL_SERVER_ERROR));
+		when(registrationProcessorRestClientService.getApi(any(), any(), anyString(), any(), any()))
+				.thenThrow(ex);
+		
+
+
+		MessageDTO result = uinGeneratorStage.process(messageDTO);
+		// assertTrue(result.getIsValid());
+	}
+	
+	@Test
+	public void deactivateTesApiResourceException() throws ApisResourceAccessException,
+			PacketManagerException,
+			IOException, JsonProcessingException, JSONException {
+		Map<String, String> fieldMap = new HashMap<>();
+		fieldMap.put("UIN", "123456");
+		fieldMap.put("name", "mono");
+		fieldMap.put("email", "mono@mono.com");
+
+		List<String> defaultFields = new ArrayList<>();
+		defaultFields.add("name");
+		defaultFields.add("dob");
+		defaultFields.add("gender");
+		defaultFields.add("UIN");
+
+		when(utility.getDefaultSource(any(), any())).thenReturn("reg_client");
+		when(packetManagerService.getFieldByMappingJsonKey(anyString(), anyString(), any(), any())).thenReturn("0.1");
+		when(packetManagerService.getFields(any(), any(), any(), any())).thenReturn(fieldMap);
+		when(idSchemaUtil.getDefaultFields(anyDouble())).thenReturn(defaultFields);
+
+		when(idSchemaUtil.getDefaultFields(anyDouble())).thenReturn(defaultFields);
+		MessageDTO messageDTO = new MessageDTO();
+		messageDTO.setRid("10031100110005020190313110030");
+		messageDTO.setReg_type(RegistrationType.valueOf("DEACTIVATED"));
+		IdResponseDTO responsedto = new IdResponseDTO();
+		String idJson = "{\"identity\":{\"IDSchemaVersion\":1.0,\"UIN\":\"4215839851\"}}";
+		InputStream idJsonStream1 = new ByteArrayInputStream(idJson.getBytes(StandardCharsets.UTF_8));
+
+		IdResponseDTO idResponseDTO = new IdResponseDTO();
+		ResponseDTO responseDTO = new ResponseDTO();
+		responseDTO.setEntity("https://dev.mosip.io/idrepo/v1.0/identity/203560486746");
+		idResponseDTO.setErrors(null);
+		idResponseDTO.setId("mosip.id.update");
+		responseDTO.setStatus("DEACTIVATED");
+		idResponseDTO.setResponse(null);
+		idResponseDTO.setResponsetime("2019-03-12T06:49:30.779Z");
+		idResponseDTO.setVersion("1.0");
+
+		// Mockito.when(packetReaderService.getFile("10031100110005020190313110030",
+		//// PacketFiles.ID.name(), "id"))//.thenReturn(idJsonStream1);
+		ApisResourceAccessException ex=new ApisResourceAccessException("");
+		when(registrationProcessorRestClientService.getApi(any(), any(), anyString(), any(), any()))
+				.thenThrow(ex);
+		
+
+
+		MessageDTO result = uinGeneratorStage.process(messageDTO);
+		// assertTrue(result.getIsValid());
+	}
+	
+	@Test
+	public void deactivateTestAlreadyDeactivated() throws ApisResourceAccessException,

```

Comment:
In this condition it is expected to reprocess or mark it failed?

Reply:
It is marked to be failed.